src/compiler/types.ts

```diff
@@ -363,7 +363,8 @@ namespace ts {
         JSDocVariadicType,
         JSDocComment,
         JSDocTag,
-        JSDocAugmentsTag,
+        JSDocAugmentsOrExtendsTag,

```

Comment:
I think we should keep `JSDocAugmentsTag` and just let `@extends` be a silent synonym for it in the parser. I don't think it's bad, for example, if the roundtrip emit converts `@extends` to `@augment`. And that's the worst side-effect I can think of.